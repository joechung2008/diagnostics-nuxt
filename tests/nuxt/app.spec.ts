import { mountSuspended, renderSuspended } from "@nuxt/test-utils/runtime";
import { afterEach, describe, expect, it, vi } from "vitest";
import App from "../../app/app.vue";
import type { Diagnostics } from "../../app/types";

describe("app/app.vue", () => {
  const originalFetch = global.fetch;

  afterEach(() => {
    // Restore original fetch after each test
    global.fetch = originalFetch;
  });

  it("should mount", async () => {
    const component = await mountSuspended(App);
    expect(component).toBeDefined();
  });

  it("should render initial state with mocked Diagnostics", async () => {
    // Mock Diagnostics object
    const mockDiagnostics: Diagnostics = {
      buildInfo: {
        buildVersion: "1.0.0-test",
      },
      extensions: {
        testExtension: {
          extensionName: "Test Extension",
          config: {
            enabled: "true",
            version: "1.0",
          },
        },
      },
      serverInfo: {
        deploymentId: "test-deployment-id",
        extensionSync: {
          totalSyncAllCount: 5,
        },
        hostname: "test-hostname",
        nodeVersions: "18.17.0",
        serverId: "test-server-id",
        uptime: 3600,
      },
    };

    // Mock the global fetch function to return the mocked Diagnostics
    const mockFetch = vi.fn().mockResolvedValue({
      json: vi.fn().mockResolvedValue(mockDiagnostics),
    });

    // Replace global fetch with the mock
    global.fetch = mockFetch;

    const { html } = await renderSuspended(App);

    expect(html()).toMatchSnapshot();
  });

  it("should render initial state with paasserverless extension", async () => {
    // Mock Diagnostics object with paasserverless extension
    const mockDiagnostics: Diagnostics = {
      buildInfo: {
        buildVersion: "1.0.0-paas",
      },
      extensions: {
        paasserverless: {
          extensionName: "PaaS Serverless Extension",
          config: {
            provider: "azure",
            runtime: "node.js",
          },
        },
        websites: {
          extensionName: "Websites Extension",
          config: {
            enabled: "true",
          },
        },
      },
      serverInfo: {
        deploymentId: "paas-deployment-id",
        extensionSync: {
          totalSyncAllCount: 10,
        },
        hostname: "paas-hostname",
        nodeVersions: "20.0.0",
        serverId: "paas-server-id",
        uptime: 7200,
      },
    };

    // Mock the global fetch function to return the mocked Diagnostics
    const mockFetch = vi.fn().mockResolvedValue({
      json: vi.fn().mockResolvedValue(mockDiagnostics),
    });

    // Replace global fetch with the mock
    global.fetch = mockFetch;

    const { html } = await renderSuspended(App);

    expect(html()).toMatchSnapshot();
  });
});
