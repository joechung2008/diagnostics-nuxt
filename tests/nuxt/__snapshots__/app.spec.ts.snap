// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`app/app.vue > should render initial state with mocked Diagnostics 1`] = `
"<div id="test-wrapper">
  <div class="flex flex-col h-screen transition-opacity opacity-0">
    <div class="relative inline-flex -space-x-px p-1"><button type="button" id="reka-dropdown-menu-trigger-v-0-0-0" aria-haspopup="menu" aria-expanded="false" data-state="closed" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] px-2.5 py-1.5 text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated active:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted"><span class="iconify i-lucide:chevron-down shrink-0 size-5" aria-hidden="true"></span>Public Cloud
        <!--v-if-->
      </button>
      <!---->
      <!--teleport start-->
      <!--teleport end-->
      <!--v-if--><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] px-2.5 py-1.5 text-sm gap-1.5 text-inverted bg-primary hover:bg-primary/75 active:bg-primary/75 disabled:bg-primary aria-disabled:bg-primary focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary">
        <!--v-if-->websites
        <!--v-if-->
      </button>
    </div>
    <div dir="ltr" data-orientation="horizontal" class="flex items-center gap-2 flex-col" justify="start">
      <div class="relative flex p-1 group border-default w-full border-b -mb-px" tabindex="0" data-orientation="horizontal" dir="ltr" style="outline-color: none; outline-style: none; outline-width: initial;" role="tablist" aria-orientation="horizontal">
        <div style="--reka-tabs-indicator-size: 0px; --reka-tabs-indicator-position: 0px;" class="absolute transition-[translate,width] duration-200 rounded-full left-0 w-(--reka-tabs-indicator-size) translate-x-(--reka-tabs-indicator-position) -bottom-px h-px bg-primary"></div><button class="group relative inline-flex items-center min-w-0 data-[state=inactive]:text-muted hover:data-[state=inactive]:not-disabled:text-default font-medium rounded-md disabled:cursor-not-allowed disabled:opacity-75 transition-colors focus:outline-none justify-center px-3 py-1.5 text-sm gap-1.5 data-[state=active]:text-primary focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-primary" data-reka-collection-item="" tabindex="-1" data-orientation="horizontal" data-active="" id="reka-tabs-v-0-0-1-trigger-extensions" role="tab" type="button" aria-selected="true" aria-controls="reka-tabs-v-0-0-1-content-extensions" data-state="active">
          <!--v-if--><span class="truncate">Extensions</span>
          <!--v-if-->
        </button><button class="group relative inline-flex items-center min-w-0 data-[state=inactive]:text-muted hover:data-[state=inactive]:not-disabled:text-default font-medium rounded-md disabled:cursor-not-allowed disabled:opacity-75 transition-colors focus:outline-none justify-center px-3 py-1.5 text-sm gap-1.5 data-[state=active]:text-primary focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-primary" data-reka-collection-item="" tabindex="-1" data-orientation="horizontal" id="reka-tabs-v-0-0-1-trigger-build" role="tab" type="button" aria-selected="false" aria-controls="reka-tabs-v-0-0-1-content-build" data-state="inactive">
          <!--v-if--><span class="truncate">Build Information</span>
          <!--v-if-->
        </button><button class="group relative inline-flex items-center min-w-0 data-[state=inactive]:text-muted hover:data-[state=inactive]:not-disabled:text-default font-medium rounded-md disabled:cursor-not-allowed disabled:opacity-75 transition-colors focus:outline-none justify-center px-3 py-1.5 text-sm gap-1.5 data-[state=active]:text-primary focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-primary" data-reka-collection-item="" tabindex="-1" data-orientation="horizontal" id="reka-tabs-v-0-0-1-trigger-server" role="tab" type="button" aria-selected="false" aria-controls="reka-tabs-v-0-0-1-content-server" data-state="inactive">
          <!--v-if--><span class="truncate">Server Information</span>
          <!--v-if-->
        </button>
      </div>
      <div id="reka-tabs-v-0-0-1-content-extensions" role="tabpanel" data-state="active" data-orientation="horizontal" aria-labelledby="reka-tabs-v-0-0-1-trigger-extensions" tabindex="0" style="animation-duration: 0s;" class="focus:outline-none w-full"></div>
      <div id="reka-tabs-v-0-0-1-content-build" role="tabpanel" data-state="inactive" data-orientation="horizontal" aria-labelledby="reka-tabs-v-0-0-1-trigger-build" hidden="" tabindex="0" class="focus:outline-none w-full">
        <!--v-if-->
      </div>
      <div id="reka-tabs-v-0-0-1-content-server" role="tabpanel" data-state="inactive" data-orientation="horizontal" aria-labelledby="reka-tabs-v-0-0-1-trigger-server" hidden="" tabindex="0" class="focus:outline-none w-full">
        <!--v-if-->
      </div>
    </div>
    <div class="tab-panel">
      <div class="stack">
        <div class="extension-root">
          <!---->
          <!---->
          <nav data-collapsed="false" class="relative flex gap-1.5 [&amp;>div]:min-w-0 flex-col" aria-label="Main" data-orientation="vertical" dir="ltr" data-reka-navigation-menu="">
            <ul class="isolate min-w-0">
              <li data-state="closed" data-orientation="vertical" class="min-w-0"><button type="button" data-reka-collection-item="" class="group relative w-full flex items-center gap-1.5 font-medium text-sm before:absolute before:z-[-1] before:rounded-md focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:before:ring-inset focus-visible:before:ring-2 focus-visible:before:ring-primary flex-row px-2.5 py-1.5 before:inset-y-px before:inset-x-0 text-muted hover:text-highlighted hover:before:bg-elevated/50 transition-colors before:transition-colors">
                  <!--v-if--><span class="truncate">Test Extension<!--v-if--></span>
                  <!--v-if-->
                </button>
                <!--v-if-->
                <!--v-if-->
              </li>
            </ul>
            <!--v-if-->
            <!--v-if-->
          </nav>
        </div>
        <!--v-if-->
      </div>
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`app/app.vue > should render initial state with paasserverless extension 1`] = `
"<div id="test-wrapper">
  <div class="flex flex-col h-screen transition-opacity opacity-0">
    <div class="relative inline-flex -space-x-px p-1"><button type="button" id="reka-dropdown-menu-trigger-v-0-0-0" aria-haspopup="menu" aria-expanded="false" data-state="closed" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] px-2.5 py-1.5 text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated active:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted"><span class="iconify i-lucide:chevron-down shrink-0 size-5" aria-hidden="true"></span>Public Cloud
        <!--v-if-->
      </button>
      <!---->
      <!--teleport start-->
      <!--teleport end--><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] px-2.5 py-1.5 text-sm gap-1.5 text-inverted bg-primary hover:bg-primary/75 active:bg-primary/75 disabled:bg-primary aria-disabled:bg-primary focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary">
        <!--v-if--> paasserverless
        <!--v-if-->
      </button><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] px-2.5 py-1.5 text-sm gap-1.5 text-inverted bg-primary hover:bg-primary/75 active:bg-primary/75 disabled:bg-primary aria-disabled:bg-primary focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary">
        <!--v-if-->websites
        <!--v-if-->
      </button>
    </div>
    <div dir="ltr" data-orientation="horizontal" class="flex items-center gap-2 flex-col" justify="start">
      <div class="relative flex p-1 group border-default w-full border-b -mb-px" tabindex="0" data-orientation="horizontal" dir="ltr" style="outline-color: none; outline-style: none; outline-width: initial;" role="tablist" aria-orientation="horizontal">
        <div style="--reka-tabs-indicator-size: 0px; --reka-tabs-indicator-position: 0px;" class="absolute transition-[translate,width] duration-200 rounded-full left-0 w-(--reka-tabs-indicator-size) translate-x-(--reka-tabs-indicator-position) -bottom-px h-px bg-primary"></div><button class="group relative inline-flex items-center min-w-0 data-[state=inactive]:text-muted hover:data-[state=inactive]:not-disabled:text-default font-medium rounded-md disabled:cursor-not-allowed disabled:opacity-75 transition-colors focus:outline-none justify-center px-3 py-1.5 text-sm gap-1.5 data-[state=active]:text-primary focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-primary" data-reka-collection-item="" tabindex="-1" data-orientation="horizontal" data-active="" id="reka-tabs-v-0-0-1-trigger-extensions" role="tab" type="button" aria-selected="true" aria-controls="reka-tabs-v-0-0-1-content-extensions" data-state="active">
          <!--v-if--><span class="truncate">Extensions</span>
          <!--v-if-->
        </button><button class="group relative inline-flex items-center min-w-0 data-[state=inactive]:text-muted hover:data-[state=inactive]:not-disabled:text-default font-medium rounded-md disabled:cursor-not-allowed disabled:opacity-75 transition-colors focus:outline-none justify-center px-3 py-1.5 text-sm gap-1.5 data-[state=active]:text-primary focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-primary" data-reka-collection-item="" tabindex="-1" data-orientation="horizontal" id="reka-tabs-v-0-0-1-trigger-build" role="tab" type="button" aria-selected="false" aria-controls="reka-tabs-v-0-0-1-content-build" data-state="inactive">
          <!--v-if--><span class="truncate">Build Information</span>
          <!--v-if-->
        </button><button class="group relative inline-flex items-center min-w-0 data-[state=inactive]:text-muted hover:data-[state=inactive]:not-disabled:text-default font-medium rounded-md disabled:cursor-not-allowed disabled:opacity-75 transition-colors focus:outline-none justify-center px-3 py-1.5 text-sm gap-1.5 data-[state=active]:text-primary focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-primary" data-reka-collection-item="" tabindex="-1" data-orientation="horizontal" id="reka-tabs-v-0-0-1-trigger-server" role="tab" type="button" aria-selected="false" aria-controls="reka-tabs-v-0-0-1-content-server" data-state="inactive">
          <!--v-if--><span class="truncate">Server Information</span>
          <!--v-if-->
        </button>
      </div>
      <div id="reka-tabs-v-0-0-1-content-extensions" role="tabpanel" data-state="active" data-orientation="horizontal" aria-labelledby="reka-tabs-v-0-0-1-trigger-extensions" tabindex="0" style="animation-duration: 0s;" class="focus:outline-none w-full"></div>
      <div id="reka-tabs-v-0-0-1-content-build" role="tabpanel" data-state="inactive" data-orientation="horizontal" aria-labelledby="reka-tabs-v-0-0-1-trigger-build" hidden="" tabindex="0" class="focus:outline-none w-full">
        <!--v-if-->
      </div>
      <div id="reka-tabs-v-0-0-1-content-server" role="tabpanel" data-state="inactive" data-orientation="horizontal" aria-labelledby="reka-tabs-v-0-0-1-trigger-server" hidden="" tabindex="0" class="focus:outline-none w-full">
        <!--v-if-->
      </div>
    </div>
    <div class="tab-panel">
      <div class="stack">
        <div class="extension-root">
          <!---->
          <!---->
          <nav data-collapsed="false" class="relative flex gap-1.5 [&amp;>div]:min-w-0 flex-col" aria-label="Main" data-orientation="vertical" dir="ltr" data-reka-navigation-menu="">
            <ul class="isolate min-w-0">
              <li data-state="closed" data-orientation="vertical" class="min-w-0"><button type="button" data-reka-collection-item="" class="group relative w-full flex items-center gap-1.5 font-medium text-sm before:absolute before:z-[-1] before:rounded-md focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:before:ring-inset focus-visible:before:ring-2 focus-visible:before:ring-primary flex-row px-2.5 py-1.5 before:inset-y-px before:inset-x-0 text-muted hover:text-highlighted hover:before:bg-elevated/50 transition-colors before:transition-colors">
                  <!--v-if--><span class="truncate">PaaS Serverless Extension<!--v-if--></span>
                  <!--v-if-->
                </button>
                <!--v-if-->
                <!--v-if-->
              </li>
              <li data-state="closed" data-orientation="vertical" class="min-w-0"><button type="button" data-reka-collection-item="" class="group relative w-full flex items-center gap-1.5 font-medium text-sm before:absolute before:z-[-1] before:rounded-md focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:before:ring-inset focus-visible:before:ring-2 focus-visible:before:ring-primary flex-row px-2.5 py-1.5 before:inset-y-px before:inset-x-0 text-muted hover:text-highlighted hover:before:bg-elevated/50 transition-colors before:transition-colors">
                  <!--v-if--><span class="truncate">Websites Extension<!--v-if--></span>
                  <!--v-if-->
                </button>
                <!--v-if-->
                <!--v-if-->
              </li>
            </ul>
            <!--v-if-->
            <!--v-if-->
          </nav>
        </div>
        <!--v-if-->
      </div>
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;
